commons/all/src/main/java/org/infinispan/commons/util/ConcurrentSmallIntSet.java

```diff
@@ -50,6 +50,9 @@ class ConcurrentSmallIntSet implements IntSet {
     * @param minCapacityExclusive The implementation performs sizing to ensure values up to this can be stored
     */
    public ConcurrentSmallIntSet(int minCapacityExclusive) {
+      if (minCapacityExclusive < 1) {
+         throw new IllegalArgumentException("minCapacityExclusive (" + minCapacityExclusive + ") < 1");
+      }
```

Comment:
Zero is not a negative number; IMO `NegativeArraySizeException` does not make sense as an exception. Except if you are suggesting throwing an exception for a negative number and another just for zero.

Also, zero is an invalid value since the set `can accommodate ints in value up to {@code minCapacityExclusive - 1}` and `This set does not support negative numbers.`

Just noticed that the parameter name does not make since it makes more sense to be named `maxCapacityExclusive`. Any objections?

Reply:
None, all good.